queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,63 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
+
+
+;; Rule imports
+(rule_import
+  "use" @include
+  "rule" @include
+  "from" @include
+  "as"? @include
+  "with"? @include
+)
+
+
+;; Rule inheritance
+(rule_inheritance
+  "use" @keyword
+  "rule" @keyword
+  "with"? @keyword
+)
+
+
+;; Wildcard names
+;; as @field for consistency with attributes in wildcard interpolations
+(wildcard (identifier) @field)
+
+
+;; Keywords in wildcard interpolations
+(
+  (identifier) @keyword
+  (#has-ancestor? @keyword "directive")
+  (#has-ancestor? @keyword "block")
+  (#any-of? @keyword
+      "config"
+      "input"
+      "log"
+      "output"
+      "params"
+      "resources"
+      "threads"
+      "wildcards"
+   )
+)
+
+(
+  (identifier) @keyword
+  (#has-ancestor? @keyword "wildcard")
+  (#any-of? @keyword
+      "config"
+      "input"
+      "log"
+      "output"
+      "params"
+      "resources"
+      "threads"
+      "wildcards"
+   )
+)
```

Comment:
```suggestion
((wildcard (identifier) @keyword)
  (#any-of? @keyword "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))

((wildcard (attribute object: (identifier) @keyword))
  (#any-of? @keyword "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
```

Reply:
Thanks, this simplifies it for the `shell` blocks. Unfortunately, it doesn't catch the `run` blocks, which can contain arbitrary Python code and in which these wildcards are expanded in a non-string context (eg. no `wildcard` scope). That's why I used `has-ancestor` there.


```suggestion
((wildcard (identifier) @keyword)
  (#any-of? @keyword "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
((wildcard (attribute object: (identifier) @keyword))
  (#any-of? @keyword "config" "input" "log" "output" "params" "resources" "threads" "wildcards"))
  
(
  (identifier) @keyword
  (#has-ancestor? @keyword "directive")
  (#has-ancestor? @keyword "block")
  (#any-of? @keyword
      "config"
      "input"
      "log"
      "output"
      "params"
      "resources"
      "threads"
      "wildcards"
   )
)
```

Reply:
Can I get a code sample

Reply:
Note: `#has-ancestor?` does a full tree traversal and is pretty expensive; use sparingly, if at all. (`#has-parent?` is somewhat cheaper but still not free.)

Reply:
See lower part in the screenshot here: https://github.com/nvim-treesitter/nvim-treesitter/pull/4302#discussion_r1188321045

Here is another example with more nesting (hence `#has-ancestor?` instead of `#has-parent?`). 
```
# Do something pointless with text files.
rule a:
    input:
        a = "a.txt",
        b = "b.txt"
    output: "out.txt"
    params:
        x_times = 5
    run:
        with open(output[0], "w") as outf:
            for i in range(params.x_times):
                with open(input.a, "r") as in_a, open(input.b, "r") as in_b:
                    outf.write(f"\n -- Round {i + 1} / {params.x_times}\n")
                    for line_a in in_a:
                        for line_b in in_b:
                            outf.write(line_b + line_a[::-1])
```
![run_example](https://github.com/nvim-treesitter/nvim-treesitter/assets/8322329/0fd1e74f-61a0-4fc1-a39f-776c6dba139d)

Reply:
Unfortunately, I still don't see how this can be solved another way.